### ISSUE TITLE
Release: 2.13.0

### ISSUE THREAD

**Issue by JorTurFer** [kedacore/keda#5275]
This issue template is used to track the rollout of a new KEDA version.

For the full release process, we recommend reading [this document](https://github.com/kedacore/keda/blob/main/RELEASE-PROCESS.md).

## Required items

- [x] https://github.com/kedacore/keda/issues/5261
- [x] https://github.com/kedacore/keda/issues/5248
- [x] https://github.com/kedacore/keda/issues/4628
- [x] https://github.com/kedacore/keda/issues/4134
- [x] https://github.com/kedacore/keda/issues/5271
- [x] https://github.com/kedacore/keda/issues/5011
- [x] https://github.com/kedacore/keda/issues/4802
- [x] https://github.com/kedacore/keda/issues/5276
- [x] https://github.com/kedacore/keda/pull/5277
- [x] https://github.com/kedacore/keda/pull/5346
- [x] https://github.com/kedacore/keda/pull/4913

## Nice to have

- [x] https://github.com/kedacore/keda/pull/5373
- [x] https://github.com/kedacore/keda/pull/5377

## Not needed
- [ ] https://github.com/kedacore/keda/issues/5371
- [ ] https://github.com/kedacore/keda/issues/2633
~- [ ] https://github.com/kedacore/keda/pull/5319~

# Timeline

We aim to release this release in the week of January 11.

## Progress

- [x] Prepare changelog (@wozniakjan)
- [x] [Welcome message supported versions](https://github.com/kedacore/keda/blob/main/pkg/util/welcome.go#L29-L30) are up-to-date (@wozniakjan)
- [x] Add the new version to [GitHub Bug report](https://github.com/kedacore/keda/blob/main/.github/ISSUE_TEMPLATE/3_bug_report.yml) template (@wozniakjan)
- [x] Create KEDA release (@tomkerkhove)
- [x] Publish new documentation version (@wozniakjan)
- [x] Setup continuous container scanning with Snyk (@tomkerkhove)
- [x] Prepare & ship Helm chart (@wozniakjan)
- [x] Prepare next release (@tomkerkhove)
- [x] Provide update in Slack (@tomkerkhove)
- [x] Tweet about new release (@tomkerkhove)

**Comment by tomkerkhove** [comment]
Added the blocker ones to the 2.13 milestone of our roadmap with high prio

**Comment by wozniakjan** [comment]
quick update regarding some of the items

> - [ ] Prepare changelog (@wozniakjan)
> - [ ] Add the new version to GitHub Bug report template (@wozniakjan)

intending to tackle in https://github.com/kedacore/keda/pull/5360

> Welcome message supported versions are up-to-date (@wozniakjan)

handled by @tomkerkhove already in https://github.com/kedacore/keda/pull/5018

**Comment by tomkerkhove** [comment]
Holding off creating a release until these are merged:
- https://github.com/kedacore/keda/pull/5373
- https://github.com/kedacore/keda/pull/5377

**Comment by zroubalik** [comment]
There are open Helm chart PRs that we should consider for the release: 
- https://github.com/kedacore/charts/pull/555
- https://github.com/kedacore/charts/pull/558
- https://github.com/kedacore/charts/pull/556
- https://github.com/kedacore/charts/pull/584

@JorTurFer @tomkerkhove @wozniakjan FYI ^

**Comment by JorTurFer** [comment]
I've just reviewed them (because sadly I was the blocker on almost all of them).
Only the cilium one is left

**Comment by tomkerkhove** [comment]
Tag created, build ongoing: https://github.com/kedacore/keda/actions/runs/7580118918

**Comment by JorTurFer** [comment]
Should we include this PR https://github.com/kedacore/keda/pull/5407?
a user requested it by slack: https://kubernetes.slack.com/archives/CKZJ36A5D/p1705601623453299
 WDYT @tomkerkhove ? @zroubalik ?

**Comment by wozniakjan** [comment]
docs PR freshly rebased to include latest merged commits, ptal 
- https://github.com/kedacore/keda-docs/pull/1290

**Comment by wozniakjan** [comment]
> Should we include this PR #5407?

given the tag has been created and build is ongoing, my gut feeling is #5407 will have to wait for next release...

**Comment by wozniakjan** [comment]
I tried to prepare the helm release as well, ptal
* https://github.com/kedacore/charts/pull/593

**Comment by tomkerkhove** [comment]
> Should we include this PR #5407? a user requested it by slack: [kubernetes.slack.com/archives/CKZJ36A5D/p1705601623453299](https://kubernetes.slack.com/archives/CKZJ36A5D/p1705601623453299) WDYT @tomkerkhove ? @zroubalik ?

I did not wait for it since we did not identify as a blocking feature, sorry =)

**Comment by tomkerkhove** [comment]
Release notes and discussion updated with changelog: https://github.com/kedacore/keda/releases/tag/v2.13.0

**Comment by tomkerkhove** [comment]
Hotfix branch: https://github.com/kedacore/keda/tree/release/v2.13

**Comment by tomkerkhove** [comment]
We should be good to go, he only remaining aspect is Helm chart + announcement which I will do once chart is ready.

@wozniakjan feel free to take your time, I'm on CNCF slack if you need me.

**Comment by wozniakjan** [comment]
thanks @tomkerkhove, I believe I have clarity regarding both [CRD diff](https://github.com/kedacore/charts/commit/949b54f6fb4e72804beada6b2d71184f10174b13) as well as [webhook disablement option](https://github.com/kedacore/charts/pull/593#discussion_r1458774953). Ptal if https://github.com/kedacore/charts/pull/593 is ready to merge.

**Comment by tomkerkhove** [comment]
Helm is merged and published release notes

**Comment by JorTurFer** [comment]
And I think that it's time for the 1st hotfix:
```bash
2024/01/19 11:51:06 maxprocs: Updating GOMAXPROCS=1: using minimum allowed GOMAXPROCS
2024-01-19T11:51:06Z	INFO	setup	Starting manager
2024-01-19T11:51:06Z	INFO	setup	KEDA Version: 2.13.0
2024-01-19T11:51:06Z	INFO	setup	Git Commit: 367fcd32f1e72be504ada53165d07c33d75fc0f7
2024-01-19T11:51:06Z	INFO	setup	Go Version: go1.21.6
2024-01-19T11:51:06Z	INFO	setup	Go OS/Arch: linux/amd64
2024-01-19T11:51:06Z	INFO	setup	Running on Kubernetes 1.27	{"version": "v1.27.7"}
2024-01-19T11:51:07Z	INFO	starting server	{"kind": "health probe", "addr": "[::]:8081"}
I0119 11:51:07.055862       1 leaderelection.go:250] attempting to acquire leader lease keda/operator.keda.sh...
2024-01-19T11:51:07Z	INFO	controller-runtime.metrics	Starting metrics server
2024-01-19T11:51:07Z	INFO	controller-runtime.metrics	Serving metrics server	{"bindAddress": ":8080", "secure": false}
E0119 11:51:24.388650       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:27.052988       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:30.112698       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:34.269628       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:38.482921       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:41.628618       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:45.826308       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:49.208734       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"
E0119 11:51:52.240208       1 leaderelection.go:369] Failed to update lock: leases.coordination.k8s.io "operator.keda.sh" is forbidden: User "system:serviceaccount:keda:keda-operator" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "keda"

```

**Comment by tomkerkhove** [comment]
Please open an issue on the Helm repo for this

**Comment by JorTurFer** [comment]
Yeah, I'm trying to figure out why this passed e2e test on KEDA side but it's failing using helm

**Comment by tomkerkhove** [comment]
Looks like an RBAC issue in the Helm chart

**Comment by tomkerkhove** [comment]
- Tweet: https://twitter.com/kedaorg/status/1748312576306368918
- LinkedIn: https://www.linkedin.com/in/tomkerkhove?miniProfileUrn=urn%3Ali%3Afs_miniProfile%3AACoAAAJJU8UBrZxIUu9yAB3KM48Xr4AQqVeAEzc

**Comment by tomkerkhove** [comment]
The release is done 🎉 Thanks for the valuable help, @wozniakjan!

**Comment by tomkerkhove** [comment]
> Please open an issue on the Helm repo for this

Let's follow-up there, likely this requires Helm hotfix only
